src/marine/copernicus_l4icethk2ioda.py

```diff
@@ -0,0 +1,147 @@
+
+import ioda_conv_engines as iconv
+from collections import defaultdict, OrderedDict
+from orddicts import DefaultOrderedDict
+
+locationKeyList = [
+    ("latitude", "float"),
+    ("longitude", "float"),
+    ("datetime", "string"),
+]
+
+obsvars = {
+    'sith': 'sith',
+}
+
+AttrData = {
+    'converter': os.path.basename(__file__),
+    'nvars': np.int32(len(obsvars)),
+}
+
+DimDict = {
+}
+
+VarDims = {
+    'sith': ['nlocs'],
+}
+
+
+class copernicus(object):
+    def __init__(self, filename):
+        # Create a simple data structure for Copernicus L4 sith product obtained from
+        # https://cds.climate.copernicus.eu/cdsapp#!/dataset/satellite-sea-level-global?tab=form
+        ncd = nc.Dataset(filename, 'r')
+        lons = ncd.variables['lon'][:].ravel()
+        lats = ncd.variables['lat'][:].ravel()
+        sith = np.squeeze(ncd.variables['sea_ice_thickness'][:]).ravel()
+        err = np.squeeze(ncd.variables['uncertainty'][:]).ravel()
+        self.date = ncd.getncattr('time_coverage_start')
+        ncd.close()
+        # masked out the Nan values
+        sithdata = {'lon': lons, 'lat': lats, 'sith': sith, 'err': err}
+        df = pd.DataFrame(sithdata)
+        df2 = df.dropna().reset_index(drop=True)
+        self.lons = df2['lon']
+        self.lats = df2['lat']
+        self.sith = df2['sith']
+        self.err = df2['err']
+        self.nlocs = len(self.sith)
+        # Same time stamp for all obs within 1 file
+        self.datetime = np.empty_like(self.sith, dtype=object)
+        self.datetime[:] = self.date
+
+
+class copernicus_l4icethk2ioda(object):
+
+    def __init__(self, filename, datetime=None):
+        self.filename = filename
+        self.datetime = datetime
+        self.varDict = defaultdict(lambda: defaultdict(dict))
+        self.metaDict = defaultdict(lambda: defaultdict(dict))
+        self.outdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        self.var_mdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        self._read()
+
+    # Open input file and read relevant info
+    def _read(self):
+        # set up variable names for IODA
+        iodavar = 'sea_ice_thickness'
+        self.varDict[iodavar]['valKey'] = iodavar, iconv.OvalName()
+        self.varDict[iodavar]['errKey'] = iodavar, iconv.OerrName()
+        self.varDict[iodavar]['qcKey'] = iodavar, iconv.OqcName()
+        self.var_mdata[iodavar, iconv.OvalName()]['units'] = 'm'
+        self.var_mdata[iodavar, iconv.OerrName()]['units'] = 'm'
+        # read input filename
+        sith = copernicus(self.filename)
+        # put the time at 00 between start and end coverage time
+        if self.datetime is not None:
+            ymdh = datetime.strptime(self.datetime, "%Y%m%d%H")
+            ymdhm = ymdh.strftime("%Y-%m-%dT%H:%M:%SZ")
+            sith.datetime[:] = ymdhm
+        # map copernicus to ioda data structure
+        self.outdata[('datetime', 'MetaData')] = sith.datetime
```

Comment:
@kbhargava I think this is what is test error is pointing to. The ioda_conv_engines.py library expects "dateTime" instead of "datetime".

On a higher level, it looks like this script is still using the old names instead of the new OBS Convention names. For example "Location" should be used now instead of "nlocs" for DimDict on line 112 below. It would be good to check for usage of OBS Conventional names throughout the script.

Reply:
@kbhargava I also run ctest with a new build. I got the same error. ```Required variable 'MetaData/dateTime' does not exist.``` 
We have fixed the script based on new naming convention.

Reply:
you may already have this correct, but note the new `dateTime` is a long integer which is seconds from epoch with that typically being 01Jan1970

the old `datetime` is a string

Reply:
@srherbener thank for your very prompt response as always, stupid enough I assumed the converor was following new convention and still running into the issue. Note to self check stuff like this before pinging you.  @BenjaminRuston thank for pointing this out. @mjhossen let me know if you need any help transitioning to the new OBS Convention names.

Reply:
@kbhargava updated the script with new naming convention. ctest passed.

Reply:
@kbhargava no worries about asking me for help. Sometimes the best way to find something is to have someone else take a look.